src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -288,7 +288,10 @@ private static function hasUserChanged(UserInterface $originalUser, TokenInterfa
         }
 
         if ($originalUser instanceof PasswordAuthenticatedUserInterface || $refreshedUser instanceof PasswordAuthenticatedUserInterface) {
-            if (!$originalUser instanceof PasswordAuthenticatedUserInterface || !$refreshedUser instanceof PasswordAuthenticatedUserInterface || $originalUser->getPassword() !== $refreshedUser->getPassword()) {
+            if (!$originalUser instanceof PasswordAuthenticatedUserInterface
+                || !$refreshedUser instanceof PasswordAuthenticatedUserInterface
+                || $refreshedUser->getPassword() !== ($originalUser->getPassword() ?? $refreshedUser->getPassword())
```

Comment:
Here is the meat of this PR: ignore null passwords from the session storage

Reply:
Wouldn't this also mean that you lose the ability to detect that the user changed the password?

Reply:
Absolutely. See the main thread of this PR for some thoughts on the topic.